### PR TITLE
Remove UploadWorkManager

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

**Description**

- Remove `UploadWorkManager` and its only implementation `SquashedChangesUploadWorkManager`
- Inlined `UploaderImpl` into `Uploader` which is no longer an interface
- Move `Downloader` to the `download` package

**Alternative(s) considered**
NA

**Type**
Code health

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: engine/src/main/java/com/google/android/fhir/sync/Config.kt [hunk 1]
```diff
@@ -40,7 +40,6 @@ val defaultRetryConfiguration =
 object SyncDataParams {
   const val SORT_KEY = "_sort"
   const val LAST_UPDATED_KEY = "_lastUpdated"
-  const val ADDRESS_COUNTRY_KEY = "address-country"
```
Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/sync/Config.kt:43]: this is just an unused variable.

**Review by anchita-g — APPROVED** [review]
LGTM!

File: engine/src/main/java/com/google/android/fhir/sync/upload/Uploader.kt [hunk 2]
```diff
@@ -18,19 +18,83 @@ package com.google.android.fhir.sync.upload
...
+ * Implementation of the [Uploader]. It orchestrates the pre processing of [LocalChange] and
+ * constructing appropriate upload requests via [UploadWorkManager] and uploading of requests via
+ * [DataSource]. [Uploader] clients should call upload and listen to the various states emitted by
+ * [UploadWorkManager] as [UploadState].
+ */
+internal class Uploader(
+  private val dataSource: DataSource,
+) {
+  private val patchGenerator = PerResourcePatchGenerator
+  private val requestGenerator = TransactionBundleGenerator.getDefault()
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply engine/src/main/java/com/google/android/fhir/sync/upload/Uploader.kt:44]: yes, but that will happen in a follow up PR.

File: engine/src/main/java/com/google/android/fhir/sync/upload/Uploader.kt [hunk 3]
```diff
@@ -18,19 +18,83 @@ package com.google.android.fhir.sync.upload
...
 
   /**
    * Uploads the local changes to the [DataSource]. Particular implementations should take care of
    * transforming the [LocalChange]s to particular network operations. If [ProgressCallback] is
    * provided it also reports the intermediate progress
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply engine/src/main/java/com/google/android/fhir/sync/upload/Uploader.kt:49]: i removed this

File: engine/src/main/java/com/google/android/fhir/sync/upload/Uploader.kt [hunk 4]
```diff
@@ -18,19 +18,83 @@ package com.google.android.fhir.sync.upload
...
    */
-  suspend fun upload(localChanges: List<LocalChange>): Flow<UploadState>
+  suspend fun upload(localChanges: List<LocalChange>): Flow<UploadState> = flow {
+    val patches = patchGenerator.generate(localChanges)
+    val requests = requestGenerator.generateUploadRequests(patches)
+    val token = LocalChangeToken(localChanges.flatMap { it.token.ids })
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply engine/src/main/java/com/google/android/fhir/sync/upload/Uploader.kt:54]: i think this is clearer (for now) to lay out the steps.

**Review by omarismail94 — APPROVED** [review]

**Pull request merged**